### ISSUE TITLE
Write/Edit necessary CMake files for testing the openworld structs

### ISSUE THREAD

**Issue by edward3-uchi** [uchicago-cs/chiventure#602]
Need tests for gencontext code and haven't implemented any proper CMake files either. These tests will test the gencontext_t struct, the roomspec_t struct, and the roomSpecList_t.

**Comment by bryan22lee** [comment]
Renamed sample_generation.h and sample_generation.c to [autogeneration.h](https://github.com/uchicago-cs/chiventure/blob/openworld/sample-generation/include/openworld/autogenerate.h) and [autogeneration.c](https://github.com/uchicago-cs/chiventure/blob/openworld/sample-generation/src/openworld/src/autogenerate.c) and moved the source code to the [src/openworld/src/](https://github.com/uchicago-cs/chiventure/tree/openworld/sample-generation/src/openworld/src) directory. This module uses the context structs defined in [gen_structs.h](https://github.com/uchicago-cs/chiventure/blob/openworld/sample-generation/include/openworld/gen_structs.h) by @edward3-uchi to determine if a game's current room has any paths and if not, generates and adds a generated room (from the context struct) to the game. So far the [tests](https://github.com/uchicago-cs/chiventure/blob/openworld/sample-generation/tests/openworld/test_autogenerate.c) for any_paths succeed and everything compiles. Next steps are to finish writing tests for the other three implemented functions in [autogenerate.h](https://github.com/uchicago-cs/chiventure/blob/openworld/sample-generation/include/openworld/autogenerate.h) and ensure they pass. We want to make sure that the functions correctly manipulate the [gen_structs.h](https://github.com/uchicago-cs/chiventure/blob/openworld/sample-generation/include/openworld/gen_structs.h) structs to create rooms based on context.

**Comment by bryan22lee** [comment]
New branch called [openworld/autogen](https://github.com/uchicago-cs/chiventure/tree/openworld/autogen) was opened, where a [CMakeLists.txt](https://github.com/uchicago-cs/chiventure/blob/openworld/autogen/tests/openworld/test_autogenerate.c) file was edited for the testing of [test_autogenerate.c](https://github.com/uchicago-cs/chiventure/blob/openworld/autogen/tests/openworld/test_autogenerate.c) tests in the [tests/openworld/](https://github.com/uchicago-cs/chiventure/tree/openworld/autogen/tests/openworld) directory. See issue #543 for more details. [test_autogenerate.c](https://github.com/uchicago-cs/chiventure/blob/openworld/autogen/tests/openworld/test_autogenerate.c) tests are currently dependent on PR approval of [gen_structs.h](https://github.com/uchicago-cs/chiventure/blob/openworld/autogen/include/openworld/gen_structs.h) module or direct access to source code for the gen_structs module (since the [autogenerate](https://github.com/uchicago-cs/chiventure/blob/openworld/autogen/include/openworld/autogenerate.h) module mainly depends on working context structs).

**Comment by bryan22lee** [comment]
The [gen_structs](https://github.com/uchicago-cs/chiventure/blob/openworld/gen-structs/include/openworld/gen_structs.h) module was updated and a new Pull Request was made. Updated structs for the context can be seen in this module. [autogenerate.h](https://github.com/uchicago-cs/chiventure/blob/openworld/autogen/include/openworld/autogenerate.h) module was updated to reflect these changes as well. Currently, the doubly linked list structure of speclist_t in [gen_structs](https://github.com/uchicago-cs/chiventure/blob/openworld/gen-structs/include/openworld/gen_structs.h), used with DL_APPEND from [uthash.h](https://github.com/uchicago-cs/chiventure/blob/openworld/autogen/include/common/uthash.h), crashes the [test_autogenerate.c](https://github.com/uchicago-cs/chiventure/blob/openworld/autogen/src/openworld/src/autogenerate.c) test. For the next sprint, we want to begin looking into what causes this, and how we can update speclist_t to fully support [uthash.h](https://github.com/uchicago-cs/chiventure/blob/openworld/autogen/include/common/uthash.h) functions.

The CMakeLists.txt files pertaining to this issue can be found at [src/openworld/CMakeLists.txt](https://github.com/uchicago-cs/chiventure/blob/openworld/autogen/src/openworld/CMakeLists.txt), [tests/openworld/CMakeLists.txt](https://github.com/uchicago-cs/chiventure/blob/openworld/autogen/tests/openworld/CMakeLists.txt) in the [openworld/autogen](https://github.com/uchicago-cs/chiventure/tree/openworld/autogen) branch, and [tests/openworld/CMakeLists.txt](https://github.com/uchicago-cs/chiventure/blob/openworld/gen-structs/tests/openworld/CMakeLists.txt) in the [openworld/gen-structs](https://github.com/uchicago-cs/chiventure/tree/openworld/gen-structs) branch. The [CMakeLists.txt](https://github.com/uchicago-cs/chiventure/blob/openworld/autogen/CMakeLists.txt) file in the root directory was updated as well to include our openworld modules when compiling.

**Comment by dwahme** [comment]
Issue Score: ✔️+

Comments:
Great job! Here are a few things you may want to improve in future issues:

The summary for this issue is too terse. A good rule of thumb is that you should be writing at least one full paragraph in your summary (see https://uchicago-cs.github.io/cmsc22000/project/policies/#a-descriptive-summary for examples). If this issue was generated from a Scrum board card, please remember that the summary you write in the Sprint meeting is meant to be a quick draft; it needs to be fleshed out when you convert the card to an issue.
